py/objcomplex.c

```diff
@@ -89,6 +89,13 @@ STATIC mp_obj_t complex_make_new(const mp_obj_type_t *type_in, size_t n_args, si
                 return args[0];
             } else {
                 // something else, try to cast it to a complex
+                mp_float_t real, imag;
+                mp_obj_t cast_obj = mp_unary_op_maybe(MP_UNARY_OP_COMPLEX, args[0]);
+                if (cast_obj != MP_OBJ_NULL && mp_obj_is_type(cast_obj, &mp_type_complex)) {
+                    mp_obj_get_complex(cast_obj, &real, &imag);
+                    return mp_obj_new_complex(real, imag);
```

Comment:
Maybe add a comment to explain why it's needed to make a deepcopy of cast_obj since it's already of the correct type?

Reply:
Not sure, but possibly could save some bytes by having the else case assiging to real, imag and use a single return. But would have to test that.. I.e comething like 

```
    mp_float_t real, imag = 0;
    ....
    mp_obj_get_complex(cast_obj, &real, &imag);
} else {
    real = mp_obj_get_float(args[0]);
}
return mp_obj_new_complex(real, imag);
```

Reply:
Is the call to `mp_unary_op_maybe()` even needed, would it be possible to just call `mp_obj_get_complex(args[0], &real, &imag)`?  That would mirror how float works in `float_make_new`